storage/csi-cloning.md

```diff
@@ -0,0 +1,112 @@
+---
+title: CSI Cloning
+authors:
+  - "@chuffman"
+reviewers:
+  - "@gnufied”
+  - “@jsafrane”
+approvers:
+  - TBD
+  - "@..."
+creation-date: 2019-09-05
+last-updated: 2019-09-25
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# csi-cloning
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs] 
+
+## Summary
+
+We want to include the upstream CSI Cloning feature in OpenShift. 
+
+## Motivation
+
+* This has been requested by both the OCS and CNV teams
+* This feature is a core functionality offered by storage vendors that we need to enable for our customers
+
+### Goals
+
+* Rebase the downstream external-provisioner to the upstream external-provisioner, based on Kubernetes 1.16.
+* Package and ship a downstream image of a sidecar including these features.
+
+### Non-Goals
+
+* This enhancement does NOT propose the addition of other types of DataSource including Populators
+* This enhancement does NOT propose support for special cases like "out of band" cloning (support for back ends that don't have Cloning features), that sort of implementation would fall under Populators.
+* This enhancement does NOT propose any ability to shrink a PVC during a Clone request (e.g. it's considered an invalid request to clone PVC-a with a size of 10Gib to a PVC with a requested size of less than 10Gib, expansion is "ok" if the driver supports it but it's not required)
+* This enhancement does NOT propose adding any ability to transfer a Clone to a different Namespace, the new PVC (Clone) will be in the same Namespace as the origin that was specified. This also means that since this is namespaced, a user can not request a Clone of a PVC that is another Namespace. A user can only request a Clone of PVCs in his or her Namespace.
+* Cloning will only be available in CSI, cloning features will NOT be added to existing in-tree plugins or Flex drivers
+* Cloning will only be available within the same storage class 
```

Comment:
I was wondering it we could drop the above, which is from the upstream KEP.

I have the feeling that this OEP should be mostly about the sidecar, not about the in-tree changes that are coming with rebase.

Reply:
I've gone ahead and removed it. We removed these Non-goals from the CSI Snapshot enhancement for the same reason.